### PR TITLE
[TASK] Refactor docheader example

### PR DESCRIPTION
**Pull request by linawolf**

### CODE REVIEW

File: Classes/Controller/BackendController.php [hunk 1]
```diff
@@ -68,6 +68,30 @@ public function addPopulateButton(ButtonBar $buttonBar): void
...
 
+    /**
+     * @param ButtonBar $buttonBar
+     */
+    public function addButtons(ButtonBar $buttonBar): void
+    {
+        $this->addPopulateButton($buttonBar);
+        $this->addShortCutButton($buttonBar);
+        $this->addReloadButton($buttonBar);
+
```
Comment (froemken) [inline Classes/Controller/BackendController.php:79]:
```suggestion
```

File: Classes/Controller/BackendController.php [hunk 2]
```diff
@@ -68,6 +68,30 @@ public function addPopulateButton(ButtonBar $buttonBar): void
         $buttonBar->addButton($populateButton);
     }
 
+    /**
+     * @param ButtonBar $buttonBar
+     */
```
Comment (froemken) [inline Classes/Controller/BackendController.php:73]:
```suggestion
```

File: Classes/Controller/BackendController.php [hunk 3]
```diff
@@ -68,6 +68,30 @@ public function addPopulateButton(ButtonBar $buttonBar): void
...
+     * @return false|string[]
+     */
+    public function getMetaInformation(): array|false
+    {
+        $permissionClause = $GLOBALS['BE_USER']->getPagePermsClause(Permission::PAGE_SHOW);
+        $metaInformation = BackendUtility::readPageAccess(
+            $this->pageUid,
+            $permissionClause,
+        );
+        return $metaInformation;
```
Comment (froemken) [inline Classes/Controller/BackendController.php:92]:
```suggestion
        return BackendUtility::readPageAccess(
            $this->pageUid,
            $permissionClause,
        );
```

File: Classes/Controller/BackendController.php [hunk 4]
```diff
@@ -68,6 +68,30 @@ public function addPopulateButton(ButtonBar $buttonBar): void
...
+    {
+        $this->addPopulateButton($buttonBar);
+        $this->addShortCutButton($buttonBar);
+        $this->addReloadButton($buttonBar);
+
+    }
+
+    /**
+     * @return false|string[]
+     */
```
Comment (froemken) [inline Classes/Controller/BackendController.php:84]: Sure, it contains additional information like it's a string array, but it isn't. `readPageAccess` returns the page record from DB. So `uid` and `pid` is `integer`. So, just `array/false` from strict type in enough.

```suggestion
```

**Review by froemken — CHANGES_REQUESTED** [review]
Can you lowercase the `String` strict type in `buildMenu`, too?

**Comment by linawolf** [comment]
About the string, I think there are more places like that, I used to write string with capital s, being used to it from Java. Lets do a mass change and not mess up other PRs. Also wondering if there is a rule for that

**Pull request merged**